packages/static-hosting/lib/static-hosting.ts

```diff
@@ -1,9 +1,11 @@
-import { Construct, CfnOutput, RemovalPolicy, StackProps, Stack} from '@aws-cdk/core';
+import { Construct, CfnOutput, RemovalPolicy, StackProps, Stack } from '@aws-cdk/core';
 import { Bucket, BucketEncryption, BlockPublicAccess } from '@aws-cdk/aws-s3';
-import { OriginAccessIdentity, CloudFrontWebDistribution, PriceClass, ViewerProtocolPolicy, SecurityPolicyProtocol, SSLMethod, Behavior, SourceConfiguration, CloudFrontWebDistributionProps } from '@aws-cdk/aws-cloudfront';
+import { OriginAccessIdentity, CloudFrontWebDistribution, PriceClass, ViewerProtocolPolicy, SecurityPolicyProtocol, SSLMethod, Behavior, SourceConfiguration, CloudFrontWebDistributionProps, LambdaFunctionAssociation, LambdaEdgeEventType } from '@aws-cdk/aws-cloudfront';
 import { HostedZone, RecordTarget, ARecord } from '@aws-cdk/aws-route53';
 import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets';
 import { User, Group, Policy, PolicyStatement, Effect } from '@aws-cdk/aws-iam';
+import { Version } from '@aws-cdk/aws-lambda';
+import { ArbitraryPathRemapFunction } from './arbitrary-path-remap';
 
 export interface StaticHostingProps {
```

Comment:
you could look to just extend the props from the parent construct so you don't need to redefine a lot of the properties that already are defined

Reply:
Not sure that this would help much? I don't think any of the properties defined in `StaticHostingProps` are properties of `Construct`